### PR TITLE
Implement tenor clef ottava bassa

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
Resolves: #19843

### CODE REVIEW

File: src/importexport/mei/internal/meiconverter.cpp [hunk 1]
```diff
@@ -599,6 +599,7 @@ engraving::ClefType Convert::clefFromMEI(const libmei::Clef& meiClef, bool& warn
         case (3): return engraving::ClefType::C3;
         case (4): return engraving::ClefType::C4;
         case (5): return engraving::ClefType::C5;
+        //case (6): return engraving::ClefType::C4_8VB;
```
Comment (cbjeukendrup) [inline src/importexport/mei/internal/meiconverter.cpp:602]: My best guess would be this:
```suggestion
        case (4):
            if (meiClef.GetDisPlace() == libmei::STAFFREL_basic_below && meiClef.GetDis() == libmei::OCTAVE_DIS_8) {
                return engraving::ClefType::C4_8VB;
            }
            return engraving::ClefType::C4;
        case (5): return engraving::ClefType::C5;
```

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp [hunk 2]
```diff
@@ -4008,6 +4008,8 @@ void MusicXMLParserPass2::clef(const QString& partId, Measure* measure, const Fr
             clef = ClefType::C5;
         } else if (line == 4) {
             clef = ClefType::C4;
+        } else if (i == -1 && line == 4) {
+            clef = ClefType::C4_8VB;
```
Comment (cbjeukendrup) [inline src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:4012]: But now this `i == -1` case won't be reached

```suggestion
        } else if (line == 4) {
            if (i == -1) {
                clef = ClefType::C4_8VB;
            } else {
                clef = ClefType::C4;
            }
```

Comment (cbjeukendrup) [reply src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:4012]: Also might be good to update line 3960 (`if (i && !(c == "F" || c == "G")) { …`)

Comment (Jojo-Schmitz) [reply src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp]: Donr

File: src/importexport/mei/internal/meiconverter.cpp (hunk 1 above)
Comment (Jojo-Schmitz) [reply src/importexport/mei/internal/meiconverter.cpp]: Done

**Review by cbjeukendrup — APPROVED** [review]
Thanks!

**Comment by Jojo-Schmitz** [comment]
Done, but there's something wrong still, that clef is more than an octave lower, more than 2 otaves even.

**Comment by Jojo-Schmitz** [comment]
Ah, found it

File: src/importexport/mei/internal/meiconverter.cpp (hunk 1 above)
Comment (Jojo-Schmitz) [reply src/importexport/mei/internal/meiconverter.cpp]: unless @lpugin has concerns here...

Comment (lpugin) [reply src/importexport/mei/internal/meiconverter.cpp]: Looks good. `Convert::clefToMEI` should be adjusted accordingly.

Comment (Jojo-Schmitz) [reply src/importexport/mei/internal/meiconverter.cpp]: Ah, missed that. Will do Sunday or Monday

Comment (Jojo-Schmitz) [reply src/importexport/mei/internal/meiconverter.cpp]: Done

**Comment by Jojo-Schmitz** [comment]
Strange vtest failure seems it uses the Tab Serif clef now rather than the Tab clef?!?
beams-20-1 diff

**Comment by Jojo-Schmitz** [comment]
I guess thise here imposes a compatibility problem:
https://github.com/musescore/MuseScore/blob/2a911e263e32160c39176c81510ed88490137511/src/engraving/style/styledef.cpp#L589
together with existing scores and their
```xml
    <tabClef>31</tabClef>
```
Which now would need to be
```xml
    <tabClef>32</tabClef>
```

**Comment by Jojo-Schmitz** [comment]
Unless I move `C4_8VB` down between these lines:
https://github.com/musescore/MuseScore/blob/8516fa0ec0a84638cc8bead92e1286975fa6f469/src/engraving/types/types.h#L543-L544

**Comment by Jojo-Schmitz** [comment]
Doesn't seem to be a good idea to store the tab clef style in the .mss files as an `int` rather than as the actual clef name line as it is in the mcsx files. I guess though it is way too late to fix this.

**Comment by Jojo-Schmitz** [comment]
Hmm, now a whjole bunch  of unit tests fail, and I have no idea why

**Comment by cbjeukendrup** [comment]
> Doesn't seem to be a good idea to store the tab clef style in the .mss files as an int rather than as the actual clef name like it is in the mcsx files. I guess though it is way too late to fix this.

Yes, that's very unfortunate. But fixing it is indeed difficult, and would require compatibility reading code for style XML, which we (surprisingly to me) don't really have at all, except for some ad-hoc checking for renamed settings. Another bit of tech debt to clean up eventually...

Anyway, this PR now looks good to me, are you happy for it to be tested and merged now?

**Review by cbjeukendrup — APPROVED** [review]

**Comment by zacjansheski** [comment]
Tested on MacOS 14, Windows 11, Ubuntu 22.04.3. Approved
https://github.com/musescore/MuseScore/issues/19843 FIXED

**Pull request merged**